ls1intum/Artemis
# [Feature] Enable Swift (with SCA) exercises for Jenkins

### Checklist

### Motivation and Context


Until now, creating Swift programming exercises was only possible in a Bamboo/Bitbucket environment. This enables the creation also on Jenkins/Gitlab environments.

### Description

A new Jenkins pipeline for Swift was created, which executes the Swift build and test command and for SCA exercises the swiftlint command.
For a local setup of Jenkins/GitLab, a new Dockerfile was created which extends the existing one by installing dependencies for Swift and SwiftLint (SCA).

The existing Dockerfile had a wrong indentation. This is fixed in this PR.

### Steps for Testing

NOTE: This PR needs to be tested on the testserver2. As it is a feature for the Jenkins/GitLab environment.

1. Log in to Artemis
2. Navigate to Course Administration
3. Create a Swift programming exercise (With and without SCA enabled)
4. Verify that the solution repo has a score of 100% (and 0 issues for SCA)
5. Verify that the template repo has a score of 0% (and issues regarding the TODOS: for SCA)
6. Verify that submitting is working as expected and SCA exercises are getting SCA feedbacks

### Test Coverage

JenkinsBuildPlanCreator.java: 95%
JenkinsProgrammingLanguageFeatureService.java: 100%
JenkinsService.java: 46%

![image](https://user-images.githubusercontent.com/33484318/103662296-0db79a80-4f70-11eb-89a9-7f57e9565904.png)

![image](https://user-images.githubusercontent.com/33484318/103662326-17d99900-4f70-11eb-9791-884f9ff5da7f.png)

![image](https://user-images.githubusercontent.com/33484318/103662370-23c55b00-4f70-11eb-9f96-087db9797c62.png)
